### ISSUE TITLE
Feature gates page mentions Docker runtime

### ISSUE THREAD

**Issue by sftim** [kubernetes/website#31379]
**This is a Bug Report**

**Problem:**

Some of the [feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) mention Docker.
Docker integration (via _dockershim_) is deprecated

**Proposed Solution:**
For the one deprecated gate where the description mentions Docker:
- `Accelerators` (deprecated)

then revise the feature gate description to explain that it provided an early form of device plugin support, that it's no longer available, and link to https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/

For each other gate name where the docs mention Docker:
- `AppArmor` (beta)
- `ExperimentalHostUserNamespaceDefaulting` (beta)

establish whether the feature gate makes sense after the dockershim is removed.
If the feature gate still makes sense, reword the description to be neutral to the container runtime. If the gate is in use but no longer makes sense, file an issue against Kubernetes itself to make sure that the gate is removed.

:information_source:  Note that in the feature gates page, deprecated really means “out of use”. Turning on any available deprecated feature gate has no effect; Kubernetes behaves as if you turned the feature off. Many “deprecated” feature gates aren't settable at all in current Kubernetes.

**Page to Update:**
https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

**Additional Information:**
/sig node

**Comment by SubhasmitaSw** [comment]
In order to proceed with the dockershim removal for the next release, the docker mentions in the [feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)  specifically 
`ExperimentalHostUserNamespaceDefaulting (beta)` shows an obsolete feature flag, see [#76982](https://github.com/kubernetes/kubernetes/issues/76982), to be able to use this particular feature, this issue must be addressed.

**Comment by chris-short** [comment]
Hey @SubhasmitaSw the last message is a little confusing to me. If they're ripping out `ExperimentalHostUserNamespaceDefaulting` at some point in the future, can we just add 'beta/to be obsoleted' to the docs or should we wait on engineers to pull the docs out at the appropriate time?  
  
cc @nate-double-u @sftim

**Comment by sftim** [comment]
`ExperimentalHostUserNamespaceDefaulting` still has this description that, yep, still mentions Docker:

> Enabling the defaulting user namespace to host. This is for containers that are using other host namespaces, host mounts, or containers that are privileged or using specific non-namespaced capabilities (e.g. `MKNODE`, `SYS_MODULE` etc.). This should only be enabled if user namespace remapping is enabled in the Docker daemon.

**Comment by sftim** [comment]
Can I check if somebody is working on this?

**Comment by sftim** [comment]
One more thing to fix, as far as I can see:
```markdown
`ExperimentalHostUserNamespaceDefaulting`: Enabling the defaulting user namespace
to host. This is for containers that are using other host namespaces, host mounts, or
containers that are privileged or using specific non-namespaced capabilities
(e.g. `MKNODE`, `SYS_MODULE` etc.). This should only be enabled if user namespace
remapping is enabled in the Docker daemon.
```

I _think_ the fix is to change “Docker daemon” to “container runtime”, but I'm not sure.

**Comment by vaibhav2107** [comment]
@SubhasmitaSw Are you still working on this issue?

**Comment by SubhasmitaSw** [comment]
@vaibhav2107 You can take it up, not having the bandwidth to come back to this. 

/unassign

**Comment by sftim** [comment]
Help is welcome.

**Comment by Aaina26** [comment]
For App Armor I think the AppArmor(beta) can be removed. This feature is still in use and is stable from this release. 
![image](https://github.com/user-attachments/assets/f077b914-fc1a-4cb7-9576-ac144e4aa422)

**Comment by Aaina26** [comment]
For ExperimentalHostUserNamespaceDefaulting feature flag, I see that there is an issue raised through which a PR to deprecate the flag has been merged. So, I think it is safe to remove it as well.

I can work on this issue @sftim  once the issue for this flag has been closed. It will be a confirmation for me. I can confirm in sig node as well if you suggest.

**Comment by Aaina26** [comment]
This issue is no longer relevant for the latest documentation as it does not mention docker anywhere. Also 2 of the feature gates mentioned have been removed. 
Documentation for version before v1.30 still mention docker daemon. If changes need to be done in previous docs then I can do it. Otherwise the issue can be closed as well.

**Comment by sftim** [comment]
/close

Nothing to fix. We rarely prioritize docs backports.
